### ISSUE TITLE
🛠 Tooling: Add eslint-plugin-n

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [JoshuaKGoldberg/template-typescript-node-package#470]
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

This template says it's for Node packages. So it makes sense that it'd add [eslint-plugin-node](https://github.com/mysticatea/eslint-plugin-node) to its list of ESLint plugins!

### Additional Info

...however, that package doesn't (yet?) have a recommended TypeScript preset: https://github.com/mysticatea/eslint-plugin-node/issues/236. I tried extending from `plugin:node/recommended` locally and received a bunch of false positive complaints.

```plaintext
...

/Users/josh/repos/template-typescript-node-package/src/setup/steps/updateLocalFiles.ts
  3:32  error  "../readFileAsJSON.js" is not found  node/no-missing-import

/Users/josh/repos/template-typescript-node-package/vitest.config.ts
  1:30  error  "vitest/config" is not found  node/no-missing-import
```

Marking as blocked on `eslint-plugin-node` providing a TypeScript-focused recommended preset.

**Comment by navin-moorthy** [comment]
This is a fort of eslint-plugin-node that is well maintained - https://github.com/eslint-community/eslint-plugin-n

**Comment by JoshuaKGoldberg** [comment]
Oh great let's go with that! Thanks for the pointer.
